src/allocators.jl

```diff
@@ -0,0 +1,40 @@
+struct WrappedModel
+    model
+    wrapped_model
+end
```

Comment:
it could be more ergonomic to store the bump allocator as a field here instead of `wrapped_model`; then `(wm::WrappedModel)(args...; kw...)` could do the 
```
        with_allocator(b) do
            inputs = Adapt.adapt(T, args)
            ret = Array(model_aa(inputs...; kw...))
            reset!(b)
            return ret
        end
```
thing

Reply:
Yeah. Done but might be improvable?